src/codegen.cpp

```diff
@@ -2080,24 +2080,20 @@ orc::ThreadSafeModule jl_create_llvm_module(StringRef name, orc::ThreadSafeConte
     // imaging mode. The structure of v4 is slightly nicer for debugging JIT
     // code.
     if (!m->getModuleFlag("Dwarf Version")) {
-        int dwarf_version = 4;
-#ifdef _OS_DARWIN_
-        if (imaging_mode)
-            dwarf_version = 2;
-#endif
+        int dwarf_version = triple.isOSDarwin() && imaging_mode ? 2 : 4;
         m->addModuleFlag(llvm::Module::Warning, "Dwarf Version", dwarf_version);
     }
     if (!m->getModuleFlag("Debug Info Version"))
         m->addModuleFlag(llvm::Module::Warning, "Debug Info Version",
             llvm::DEBUG_METADATA_VERSION);
     m->setDataLayout(DL);
     m->setTargetTriple(triple.str());
-
-#if defined(_OS_WINDOWS_) && !defined(_CPU_X86_64_) && JL_LLVM_VERSION >= 130000
     // tell Win32 to assume the stack is always 16-byte aligned,
     // and to ensure that it is 16-byte aligned for out-going calls,
     // to ensure compatibility with GCC codes
-    m->setOverrideStackAlignment(16);
+#if JL_LLVM_VERSION >= 130000
+    if (triple.isOSWindows() && triple.getArch() != Triple::x86_64)
```

Comment:
minor nit: this is meant to only apply to i686, but we were being sloppy before
```suggestion
    if (triple.isOSWindows() && triple.getArch() == Triple::x86)
```